### PR TITLE
Fixed unexpected model generation bug.

### PR DESCRIPTION
**Pull request by jiachen1120**
@NicholasAzar @stevehu 
This PR is related to API-46 issue. Look forward to your review.

- Added schema type check
- Added properties number check

RFC for this issue: https://github.com/networknt/light-rfcs/blob/master/light-codegen/0003-model-generation-bug.md

### CODE REVIEW

File: light-rest-4j/src/main/java/com/networknt/codegen/rest/OpenApiGenerator.java [hunk 1]
```diff
@@ -233,6 +233,9 @@ public void generate(String targetPath, Object model, Any config) throws IOExcep
                     String classVarName = key;
                     String modelFileName = key.substring(0, 1).toUpperCase() + key.substring(1);
                     //System.out.println("props = " + Any.wrap(props));
+                    if (!"object".equals(type)) {
+                        continue;
+                    }
```
Comment (NicholasAzar) [inline light-rest-4j/src/main/java/com/networknt/codegen/rest/OpenApiGenerator.java]: Could you add a comment to this check explaining why it's necessary

Comment (jiachen1120) [reply light-rest-4j/src/main/java/com/networknt/codegen/rest/OpenApiGenerator.java]: Sure, I will do taht

**Comment by ddobrin** [comment]
Is this change backwards compatible with the classes generated by the light-codegen at this time?

**Comment by 243826** [comment]
It looks like this fix introduces another bug where the name where the model generation for the types is suppressed but the types are not replaced with their corresponding primitive boxed types or array types. It results in non compiling code.

e.g. I had  schemas
```
"UserId": {
  "type": "string",
   "minLength" : 8,
   "maxLength: 32
},
"Pair": {
  "type": "object",
  "required": [
      "primary",
      "secondary"
  ],
  "properties": {
     "primary" : {
        "$ref" : "#/components/schemas/UserId"
      },
     "secondary" : {
        "$ref" : "#/components/schemas/UserId"
      }
  }
}
```

I would get the  following model for ```Pair```
```
class Pair {
  UserId primary;
  UserId secondary;
  
  // irrelevant details deleted
}
```

but will not get the model for ```UserId```. Obviously the code would not compile.

**Comment by jiachen1120** [comment]
@243826 Thank you for pointing this out. All properties start with `$ref` at this stage will be treated as objects to generate the corresponding code. This is incorrect and should be replaced with the type it references.

**Comment by stevehu** [comment]
@jiachen1120 @243826 There is a utility @ddobrin built to bundle multiple specification files and resolve the references in the components. I am wondering if we should try it out see what is the final spec before feeding it to the light-codegen.

**Comment by 243826** [comment]
I can give it a shot. I was thinking probably I would need something like
that soon.

Could you help with more pointers?

**Comment by stevehu** [comment]
https://github.com/networknt/openapi-bundler is the repository and document is in the readme. In addition, this document might help. https://doc.networknt.com/development/best-practices/openapi3/#guidelines

**Pull request merged**